tests/test_timefuncs.py

```diff
@@ -1035,3 +1036,18 @@ def test_timedelta_range() -> None:
 def test_dateoffset_freqstr() -> None:
     offset = DateOffset(minutes=10)
     check(assert_type(offset.freqstr, str), str)
+
+
+def timedelta64_and_arithmatic_operator() -> None:
+    s1 = pd.Series(data=pd.date_range("1/1/2020", "2/1/2020"))
+    s2 = pd.Series(data=pd.date_range("1/1/2021", "2/1/2021"))
+    s3 = s2 - s1
+    td = np.timedelta64(1, "M")
+    assert_type((s1 - td), TimestampSeries)
+    assert_type((s1 + td), TimestampSeries)
+    assert_type((s1 * td), NoReturn)  # pyright: ignore
+    assert_type((s1 / td), NoReturn)  # pyright: ignore
+    assert_type((s3 - td), TimedeltaSeries)
+    assert_type((s3 + td), TimedeltaSeries)
+    assert_type((s3 * td), NoReturn)  # pyright: ignore
```

Comment:
Can you do the following:
1. for the 4 cases where you are expecting `TimestampSeries` or `TimedeltaSeries`, use `check(assert_type(  operation, expected_type), expected_type, expected_subtype)` where operation is the operation you are testing, expected_type is the type expected, and expected_subtype is `pd.Timestamp` or `pd.Timedelta` as appropriate
2. For the 3 cases where you have `NoReturn`, can you change that to `Never` , which you will need to import from `typing_extensions`
3. The `# pyright: ignore` is correct - pyright is doing the right thing here, but mypy is not

Reply:
Sir regarding point 2 and 3 when I changed it to `Never` pyrigth shows error, mypy works ok and if we add the `#type: ignore` both work fine so should I add it.
Because in the pyright error it says `"assert_type" mismatch: expected "Never" but received "Unknown"`

Reply:
Yes, you will still need the `# pyright: ignore` where you had it before

Reply:
done sir